composer.json

```diff
@@ -21,8 +21,8 @@
         "webmozart/assert": "^1.9"
     },
     "require-dev": {
-        "laminas/laminas-coding-standard": "^2.0",
-        "maglnet/composer-require-checker": "^2.0",
+        "laminas/laminas-coding-standard": "~2.2.0",
+        "maglnet/composer-require-checker": "^2.0 || ^3.0",
```

Comment:
crc 2 supports php 7.3 and crc 3 supports php 7.4 and 8

Reply:
Not sure how to fix this:

```
  Problem 1
    - Root composer.json requires maglnet/composer-require-checker == 2.1.0.0 -> satisfiable by maglnet/composer-require-checker[2.1.0].
    - maglnet/composer-require-checker 2.1.0 requires php ^7.2 -> your php version (8.0.5) does not satisfy that requirement.
 ```
 
 @boesing Do we even need this composer-require-checker?

Reply:
> Do we even need this composer-require-checker?

I don't think so. I've ported this over from a private project where I've used CRC. As we do not have anything related to CRC in our CI pipeline, I think we can simply remove it.

But as I see many projects dropping support of PHP versions quite rough, we should overthink the way how we test locked dependencies.
Imho, it would be enough to have locked dependencies tested on the latest stable PHP version declared by laminas (which actually is PHP 7.4 afaik). Should we overthink the way the matrix generates the jobs? Having lowest, latest for all versions seems legit but having locked dependencies tested only on stable version would prevent us from having configurations where we have to exclude these "jobs". Afaik, locked versions are used for static analysis and codestyle in the 1st place. Thoughts, @weierophinney?

Reply:
I'm of two minds on this.

Having a locked run discrete from latest means that we can identify when a new version of a dependency causes breakage in our code. Additionally, for scenarios where we have more than two items in the constraint (e.g., `^1.4 || ^2.2 || ^3.0`), it allows us to actually test against each constraint. Granted, that latter is rare, and, generally speaking, going to get statistically unlikely as we narrow the PHP requirements in individual release branches.

However, it also means 33% more runs, which means slower results when running QA. But when we're only supporting 3 PHP versions at a time, that's only 3 more runs, which tends not to affect things a ton.

Might be something for the TSC?

Reply:
Can bring that up as an agenda item 👍🏼